### PR TITLE
Add export form data to CSV sample

### PR DESCRIPTION
**Pull request by datalogics-bhaugen**
#### Changes in this pull request

- adds a sample that demonstrates exporting form data to a CSV file

#### Checklist for approving this pull request

- [ ] There are **unit tests** for new/changed code, or a good explanation why this was not possible.
- [ ] All **CI builders** have indicated success (Give them a few minutes to notice the pull request.)
- [ ] The **Pull request title** has the JIRA issue numbers separated by spaces (if any), a space, and then a short, but descriptive summary.
- [ ] **Commit messages** are well formed: [A note about Git commit messages](http://www.tpope.net/node/106)
- [ ] New public packages, classes, and methods are **documented**. (Strongly consider documenting private classes and methods.)

#### Blockers

Add a checkbox for yourself with your **&#64;name** to this list if you are holding this PR open for review. PR Shepherd, please hold off merging this PR until these are all checked:

- [x] _&lt;add your name here with an **@**>_

### CODE REVIEW

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-kam** [comment]
Hm. Interesting.

CSV is trickier than you think. All kinds of trouble escaping values that have funny values, or working to meet various standards. Would you consider [Apache Commons CSV](https://commons.apache.org/proper/commons-csv/)?

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-kam** [comment]
See datalogics-bhaugen#1

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-bhaugen** [comment]
Added a test for this sample. The only thing remaining that I would like to do before this gets merged is add it to the lite package.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-bhaugen** [comment]
Trigger build

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-bhaugen** [comment]
Latest commit resolved the last PMD warning that was left, this is ready for review now!

File: src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        final PDFDocument pdfDocument = DocumentUtils.openPdfDocument(inputUrl);
+
+        final PDFInteractiveForm form = pdfDocument.getInteractiveForm();
+
+        final File outputFile = new File(outputUrl.toURI());
+        if (outputFile.exists()) {
+            Files.delete(outputFile.toPath());
+        }
+
+        final PrintWriter writer = new PrintWriter(outputFile, "UTF-8");
```
Comment (datalogics-kam) [inline src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java]: ``PrintWriter`` implements ``AutoCloseable``, so use try-with-resources to close it``
(So does ``CSVPrinter``)

File: src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        checkCsvRecordValues(recordsAsList, DEFAULT_SAMPLE_OUTPUT_CSV_DATA);
+    }
+
+    private void checkCsvRecordValues(final List<CSVRecord> records, final HashMap<String, String> expectedCsvData)
+                    throws Exception {
+        for (final CSVRecord record : records) {
+            final Map<String, String> recordAsMap = record.toMap();
+            assertEquals("Record size should match expected CSV data size", expectedCsvData.size(),
+                         recordAsMap.size());
+            assertEquals("", recordAsMap, expectedCsvData);
```
Comment (datalogics-kam) [inline src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java]: Empty string for the explanation makes me wonder if this was something planned that didn't make it.

File: src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            recordAsMap.clear();
```
Comment (datalogics-kam) [inline src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java]: This confuses me. Why clear the map if it's going away anyway?

**Review by datalogics-kam — COMMENTED** [review]
This is kind of a start; I'll take more look at it later.

Good looking sample. Nice and concise.

The only other thing is to make sure your commit messages are imperative, and I don't know how I feel about the spaces before colons in commit headings, but that's maybe just me.

File: src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java (hunk 2 above)
Comment (datalogics-bhaugen) [reply src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java]: Fixed in 5387e7d02, thanks for pointing that one out!

File: src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java (hunk 1 above)
Comment (datalogics-bhaugen) [reply src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java]: Fixed in 9bb9847

File: src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java (hunk 3 above)
Comment (datalogics-bhaugen) [reply src/test/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsvTest.java]: Also fixed in 5387e7d

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-coreyy** [comment]
LGTM, tested locally

**Pull request merged**